### PR TITLE
image/store: Use errdefs for errors

### PR DESCRIPTION
**Pull request by vvoland**
Fixes:
- https://github.com/moby/moby/issues/44290

Changed errors returned to use errdefs. The original intent was to fix the mentioned issue, but I went ahead and also changed all others errors here.

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

### CODE REVIEW

File: image/store_test.go [hunk 1]
```diff
@@ -70,6 +72,7 @@ func TestRestore(t *testing.T) {
 
 	invalidPattern := id1.Encoded()[1:6]
 	_, err = imgStore.Search(invalidPattern)
+	assert.Check(t, errdefs.IsNotFound(err), "unexpected error type %T", err)
```
Comment (thaJeztah) [inline image/store_test.go:75]: I (not too long ago) found that you can use `is.ErrorType` for this; it accepts either a _type_ or a function to check the type (which can be `errdefs.IsSOME_ERROR`);

```suggestion
	assert.Check(t, is.ErrorType(err, errdefs.IsNotFound))
```

When using that, a failure looks like this (I changed the test to make it fail, but you get the idea);

```
store_test.go:75: assertion failed: error is No such image: 40b4d (*errors.withStack), not errdefs.IsInvalidParameter
```

Before (current implementation in this PR), it would look like:

```
store_test.go:75: assertion failed: expression is false: errdefs.IsInvalidParameter(err): unexpected error type *errors.withStack
```

File: image/store_test.go [hunk 2]
```diff
@@ -70,6 +72,7 @@ func TestRestore(t *testing.T) {
...
 	assert.ErrorContains(t, err, "No such image")
```
Comment (thaJeztah) [inline image/store_test.go:76]: As we're now checking the error-type to be a "not found", I wonder if we still need to check the `No  such image`. If we _do_ want to keep it (to make sure it's a 404 related to the _image_ and not anything else), perhaps more relevant would be if the error contains the expected image-id;

```suggestion
	assert.Check(t, is.ErrorContains(err, invalidPattern))
```

Short background why I was looking;

Before we had ("proper") typed errors, there was a horrible hack in the code that did string-matching to convert errors to HTTP status codes. So at the time, a "not found" error MUST contain `No such`, so that it's converted to a `404`. We got rid of that, but still left that convention for some time (as there were (sigh) other project that used the same string matching 😬)

Basically; I _think_ I have a branch somewhere that changes them to lowercase, and/or some rephrasing, having the (capital) `No such image` already gone from the test can help for some of these.

File: image/store_test.go (hunk 1 above)
Comment (vvoland) [reply image/store_test.go:75]: Oh, that's a really good find, thanks!

File: image/store_test.go (hunk 2 above)
Comment (vvoland) [reply image/store_test.go:76]: Good catch, thanks.
Initially I wanted to get rid of the local `imageNotFoundError` helper and inline it to a single `errdefs.NotFound` call, but since the message starts with uppercase the linter was complaining about it, and I assumed that we need to keep it this way for backwards compatibility 😄

**Review by cpuguy83 — APPROVED** [review]

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!
